lib/ftp.c

```diff
@@ -1940,6 +1940,17 @@ static CURLcode ftp_state_pasv_resp(struct connectdata *conn,
 #endif
   {
     /* normal, direct, ftp connection */
+
+    /* postponed address resolution in case of tcp fastopen */
+    int no_newhost = !ftpc->newhost || ftpc->newhost[0] == '\0';
```

Comment:
`ftpc->newhost` can't be NULL here, can it?

Reply:
True - a lengthy dynamic analysis of the code after `Curl_safefree(ftpc->newhost)` was necessary to find that out.
shall I make that last change (remove the check)?

Reply:
Yes please!